From cbe9fc12a64c3ae89fd1b20e9e165aa4b76293a5 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Wed, 26 Jan 2011 12:03:17 +0000
Subject: glx: fix length of GLXGetFBConfigsSGIX

The extra length is the size of the request *minus* the size of the
VendorPrivate header, not the addition.

NOTE: This is a candidate for the 7.9 and 7.10 branches

Signed-off-by: Julien Cristau <jcristau@debian.org>
Signed-off-by: Brian Paul <brianp@vmware.com>
---
diff --git a/src/glx/glxext.c b/src/glx/glxext.c
index c5e9d05..c75c9bf 100644
--- a/src/glx/glxext.c
+++ b/src/glx/glxext.c
@@ -688,7 +688,7 @@ static GLboolean
    }
    else if (strstr(psc->serverGLXexts, "GLX_SGIX_fbconfig") != NULL) {
       GetReqExtra(GLXVendorPrivateWithReply,
-                  sz_xGLXGetFBConfigsSGIXReq +
+                  sz_xGLXGetFBConfigsSGIXReq -
                   sz_xGLXVendorPrivateWithReplyReq, vpreq);
       sgi_req = (xGLXGetFBConfigsSGIXReq *) vpreq;
       sgi_req->reqType = priv->majorOpcode;
--
cgit v0.8.3-6-g21f6
